app/Http/Controllers/DataSiswaController.php

```diff
@@ -10,72 +10,35 @@ class DataSiswaController extends Controller
 {
     public function index(Request $request)
     {
-        $siswa   = Siswa::all();
-        //$siswas  = Siswa::paginate(5);
-        //$guru2 = Siswa::latest()->get();
-        
-
-        $siswas = Siswa::where([
-            ['nama', '!=', Null],
-            [function ($query) use ($request) {
-                if (($s = $request->s)) {
-                    $query->orWhere('nama', 'LIKE', '%' . $s . '%')
-                        ->orWhere('nisn', 'LIKE', '%' . $s . '%')
-                        ->orWhere('jenis_kelamin', 'LIKE', '%' . $s . '%')
-                        ->get();
-                }
-            }]
-        ])->paginate(5);
+        $this->model   = new Siswa;
+        $siswa         = $this->model->search($request);
 
         return view('admin.siswa')->with([
-            'user' => Auth::user(),
-            'siswas' => $siswas,
-            'siswa' => $siswa,
+            'user'     => Auth::user(),
+            'siswa'    => $siswa,
         ]);
     }
 
     public function store(Request $request)
     {
-        $check = Siswa::where(['nisn' => $request->nisn])->get();
-        if($check->count()>0){
-            session()->flash('notif', array('success' => false, 'msgaction' => 'Tambah Data Gagal, Data Telah Ada!'));
-            return redirect()->route('datasiswa.search');
-        }else{
-            $request->validate([
-            'nama'  => 'required',
-            'nisn'  => 'required',
-            'jenis_kelamin' => 'required'
-            //'tingkat_kelas'  => 'required'
-        ]);
-        // simpan
-        Siswa::create($request->all());
-
-        // redirect
-        return redirect()->route('datasiswa.search')->with('success', 'Data berhasil ditambahkan!');
-        }
-        
-        
+        $this->model = new Siswa;
+        $this->model->tambah_data($request);
+        return redirect()->route('datasiswa')->with('success', 'Data berhasil ditambahkan!');
     }
 
     public function update(Request $request, $id)
     {
-        $siswa  = Siswa::find($id);
-        $input  = $request->all();
-        $siswa->fill($input)->save();
-
-        return redirect()->route('datasiswa.search')->with('success', 'Data berhasil diupdate!');
+        $this->model = new Siswa;
+        $this->model->update_data($request,$id);
+        return redirect()->route('datasiswa')->with('success', 'Data berhasil diupdate!');
     }
 
     public function destroy($id)
     {
-        $siswas     = Siswa::find($id);
-        $siswas->delete();
-
-        return redirect()->route('datasiswa.search')->with('success', 'Data berhasil dihapus!');
+        $this->model = new Siswa;
+        $this->model->delete_data($id);
+        return redirect()->route('datasiswa')->with('success', 'Data berhasil dihapus!');
```

Comment:
kok malah nyatu lagi ini @syifa-sisop udah bener" di enter
```suggestion

        return redirect()->route('datasiswa')->with('success', 'Data berhasil dihapus!');
```